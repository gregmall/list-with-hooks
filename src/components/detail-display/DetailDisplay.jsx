import React from 'react';
import PropTypes from 'prop-types';


const DetailDisplay = ({ detail }) => (
  <div>
    
    <h1 data-testid ="detail">Name: {detail.name}</h1>
    <img src={detail.image}/>
    <h2>Status: {detail.status}</h2>
    <h2>Gender: {detail.gender}</h2>
  < h2>Created: {detail.created}</h2>
    <a href="/"><button>GO BACK</button></a>
  
</div>
 
);

DetailDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  created: PropTypes.string.isRequired
};


export default DetailDisplay;
