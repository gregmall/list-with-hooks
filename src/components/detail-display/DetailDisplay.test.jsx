import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DetailDisplay from './DetailDisplay';

describe('DetailDisplay component', () => {
  afterAll(() => cleanup());
  
  it('displays DetailDisplay', () => {
    const character = {
      name: 'morty',
      image: 'www.placeholder.com',
      status: 'alive',
      gender: 'male',
      created: 'some date'
    }
    
    const { asFragment } = render(<DetailDisplay detail ={character}/>);
      expect(asFragment()).toMatchSnapshot();
      
  });
});

