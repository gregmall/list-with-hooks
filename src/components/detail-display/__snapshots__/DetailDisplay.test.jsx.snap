// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailDisplay component displays DetailDisplay 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="detail"
    >
      Name: morty
    </h1>
    <img
      src="www.placeholder.com"
    />
    <h2>
      Status: alive
    </h2>
    <h2>
      Gender: male
    </h2>
    <h2>
      Created: some date
    </h2>
    <a
      href="/"
    >
      <button>
        GO BACK
      </button>
    </a>
  </div>
</DocumentFragment>
`;
